orbit/pkg/service/service_windows.go

```diff
@@ -0,0 +1,95 @@
+//go:build windows
+// +build windows
+
+package service
+
+import (
+	"os"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/orbit/pkg/constant"
+	"github.com/fleetdm/fleet/v4/orbit/pkg/platform"
+
+	"github.com/rs/zerolog/log"
+	"golang.org/x/sys/windows/svc"
+)
+
+type windowsService struct {
+	rootDir string
+}
+
+func (m *windowsService) bestEffortShutdown(serviceRootDir string) {
+	err := platform.KillProcessByName(constant.DesktopAppExecName)
+	if err != nil {
+		log.Info().Err(err).Msg("The desktop app couldn't be killed")
+	}
+
+	if serviceRootDir != "" {
+		err = platform.KillFromPIDFile(serviceRootDir, constant.OsqueryPidfile, constant.OsquerydName)
+		if err != nil {
+			log.Info().Err(err).Msg("The osquery daemon process couldn't be killed")
+		}
+	}
+}
+
+func (m *windowsService) Execute(args []string, requests <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	// Accepted service operations
+	const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
+
+	// Expected service status update during initialization
+	changes <- svc.Status{State: svc.StartPending}
+	changes <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
+
+	// The listening loop below will keep listening for new SCM requests
+	for req := range requests {
+		switch req.Cmd {
+
+		case svc.Interrogate:
+			changes <- req.CurrentStatus
+
+		case svc.Stop, svc.Shutdown:
+
+			// Service shutdown was requested
+			// Updating the service state to indicate stop op started
+			changes <- svc.Status{State: svc.StopPending}
+
+			// Best effort tear down
+			m.bestEffortShutdown(m.rootDir)
+
+			// Updating the service state to indicate that stop finished
+			changes <- svc.Status{State: svc.Stopped}
+
+			// Dummy delay to allow the SCM to pick up the changes
+			time.Sleep(500 * time.Millisecond)
+
+			// Drastic teardown
+			os.Exit(0)
```

Comment:
I think this is probably okay for now, but we may need to find a way in the future to signal a shutdown to the other threads in the Orbit process (like if we start doing some sort of data persistence and we need to make sure that a flush completes before shutdown).

Reply:
This one is really catchy. So what's happening is that the os.Exit() call will end up generating an internal signal that will be caught by the signal handler runner [here](https://github.com/fleetdm/fleet/blob/ea68e696cbe0cb89da18c0ebb5db78aea76e0596/orbit/cmd/orbit/orbit.go#L517). This will force the signal runner `interrupt()` function to execute and return, and consequently, the `interrupt()` functions for all the runners will end up being executed. This means that Runners will end up being gracefully finished. It will be up to the runner to flush data needed on their `interrupt()` function. This behavior can be tested by instrumenting the `interrupt()` code and then calling `sc stop "Fleet osquery"`

I've added a comment on the code to clarify this out.